buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestReportLogger.groovy

```diff
@@ -1,369 +0,0 @@
-import com.carrotsearch.ant.tasks.junit4.events.IEvent
-import com.carrotsearch.ant.tasks.junit4.events.IStreamEvent
-import com.carrotsearch.ant.tasks.junit4.events.SuiteStartedEvent
-import com.carrotsearch.ant.tasks.junit4.events.TestFinishedEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedQuitEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedResultEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedStartEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedSuiteResultEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedSuiteStartedEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatedTestResultEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.ChildBootstrap
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.HeartBeatEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.PartialOutputEvent
-import com.carrotsearch.ant.tasks.junit4.events.aggregated.TestStatus
-import com.carrotsearch.ant.tasks.junit4.events.mirrors.FailureMirror
-import com.carrotsearch.ant.tasks.junit4.listeners.AggregatedEventListener
-import com.carrotsearch.ant.tasks.junit4.listeners.StackTraceFilter
-import org.apache.tools.ant.filters.TokenFilter
-import org.gradle.api.logging.LogLevel
-import org.gradle.api.logging.Logger
-import org.junit.runner.Description
-
-import java.util.concurrent.atomic.AtomicInteger
-
-import static com.carrotsearch.ant.tasks.junit4.FormattingUtils.formatDescription
-import static com.carrotsearch.ant.tasks.junit4.FormattingUtils.formatDurationInSeconds
-import static com.carrotsearch.ant.tasks.junit4.FormattingUtils.formatTime
-import static com.carrotsearch.gradle.junit4.TestLoggingConfiguration.OutputMode
-
-class TestReportLogger extends TestsSummaryEventListener implements AggregatedEventListener {
-
-    static final String FAILURE_MARKER = " <<< FAILURES!"
-
-    /** Status names column. */
-    static EnumMap<? extends TestStatus, String> statusNames;
-    static {
-        statusNames = new EnumMap<>(TestStatus.class);
-        for (TestStatus s : TestStatus.values()) {
-            statusNames.put(s,
-                    s == TestStatus.IGNORED_ASSUMPTION
-                            ? "IGNOR/A" : s.toString());
-        }
-    }
-
-    JUnit4 owner
-
-    /** Logger to write the report to */
-    Logger logger
-
-    TestLoggingConfiguration config
-
-    /** Forked concurrent JVM count. */
-    int forkedJvmCount
-
-    /** Format line for JVM ID string. */
-    String jvmIdFormat
-
-    /** Output stream that logs messages to the given logger */
-    LoggingOutputStream outStream
-    LoggingOutputStream errStream
-
-    /** A list of failed tests, if to be displayed at the end. */
-    List<Description> failedTests = new ArrayList<>()
-
-    /** Stack trace filters. */
-    StackTraceFilter stackFilter = new StackTraceFilter()
```

Comment:
I think the stack trace filter is something we want to keep. The traces can get quite long with all the additional cruft of junit/runner stuff.

Reply:
This is going to be more difficult to implement. We could filter it out of the raw logs that we spit out at the end of suites but there aren't any hooks to tweak the stacktraces that get reported to Gradle, and by proxy, into build scans.